crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -190,60 +190,55 @@ impl Rule for NoFloatingPromises {
 ///
 /// doesNotReturnPromise().then(() => {});
 /// ```
-fn is_callee_a_promise(callee: &AnyJsExpression, model: &SemanticModel) -> bool {
+fn is_callee_a_promise(callee: &AnyJsExpression, model: &SemanticModel) -> Option<bool> {
     match callee {
         AnyJsExpression::JsIdentifierExpression(ident_expr) => {
-            let Some(reference) = ident_expr.name().ok() else {
-                return false;
-            };
-
-            let Some(binding) = model.binding(&reference) else {
-                return false;
-            };
+            let reference = ident_expr.name().ok()?;
+            let binding = model.binding(&reference)?;
+            let any_js_binding_decl = binding.tree().declaration()?;
 
-            let Some(any_js_binding_decl) = binding.tree().declaration() else {
-                return false;
-            };
             match any_js_binding_decl {
                 AnyJsBindingDeclaration::JsFunctionDeclaration(func_decl) => {
-                    is_function_a_promise(&func_decl)
+                    Some(is_function_a_promise(&func_decl))
                 }
-                AnyJsBindingDeclaration::JsVariableDeclarator(js_var_decl) => {
-                    is_variable_initializer_a_promise(&js_var_decl)
-                        || is_variable_annotation_a_promise(&js_var_decl)
-                }
-                _ => false,
+                AnyJsBindingDeclaration::JsVariableDeclarator(js_var_decl) => Some(
+                    is_variable_initializer_a_promise(&js_var_decl).unwrap_or(false)
+                        || is_variable_annotation_a_promise(&js_var_decl).unwrap_or(false),
+                ),
```

Comment:
Nit: I think this could have been written more concisely as:

```rs
                AnyJsBindingDeclaration::JsVariableDeclarator(js_var_decl) =>
                    is_variable_initializer_a_promise(&js_var_decl).or_else(
                        || is_variable_annotation_a_promise(&js_var_decl)
                    ),
```

Note this would propagate the `None` state up to the caller, but that's also the point of having these functions return `Option`. Don't worry about it though, you'll get the hang of this style of Rust utilities soon enough :+1:

Reply:
Thanks for the suggestion! I'll incorporate that change in the next PR.